defi/src/adaptors/db-utils/adaptor-record.ts

```diff
@@ -176,23 +177,37 @@ export const storeAdaptorRecord = async (adaptorRecord: AdaptorRecord, eventTime
         }, (currentData ?? {}) as IRecordAdaptorRecordData),
         eventTimestamp
     }
-    const obj2StoreRemoveReserveKeys = Object.entries(obj2Store).reduce((acc, [key, value]) => {
-        if (dynamoReservedKeywords.includes(key.toUpperCase())) {
-            delete acc[key];
-            return acc
-        }
-        acc[key] = value
-        return acc
-    }, {} as IRecordAdaptorRecordData)
-
+    // const obj2StoreRemoveReserveKeys = Object.entries(obj2Store).reduce((acc, [key, value]) => {
+    //     if (dynamoReservedKeywords.includes(key.toUpperCase())) {
+    //         delete acc[key];
+    //         return acc
+    //     }
+    //     acc[key] = value
+    //     return acc
+    // }, {} as IRecordAdaptorRecordData)
+    adaptorRecord.data = obj2Store 
+    const record = adaptorRecord.getCleanAdaptorRecord()
+    const { PK, SK } = adaptorRecord.keys()
 
     try {
-        console.log("Storing", obj2StoreRemoveReserveKeys, adaptorRecord.keys())
-        await dynamodb.update({
-            Key: adaptorRecord.keys(),
-            UpdateExpression: createUpdateExpressionFromObj(obj2StoreRemoveReserveKeys),
-            ExpressionAttributeValues: createExpressionAttributeValuesFromObj(obj2StoreRemoveReserveKeys)
-        }) // Upsert like
+        console.log("Storing", adaptorRecord.keys())
+        if (record) {
+            await dynamodb.put(record.toItem())
+        }
+        if (errorData) {
```

Comment:
if theres an error on a chain, does it effectively get the chain ignored? if so might be better to rug it instead to avoid incomplete data

unsure about it tho since i havent checked rest of the code so i might be completely wrong here

Reply:
I too am unsure how the code handles this, that's why left existing behavior as is.

Reply:
i previously discussed it with tawa and told him to prevent updates if one chain was missing, so i think thats the current behaviour (previously this wasnt the case and i asked him to change it), but im unsure looking at code